### PR TITLE
feat(BREAKING): v5 migration

### PR DESCRIPTION
**Pull request by dhruvkelawala**

### CODE REVIEW

**Comment by dhruvkelawala** [comment]
CI will fail because of unusable response from starknet.js `contract.call`

**Comment by fracek** [comment]
Thank you for this. Will review it this week.

File: packages/core/src/hooks/receipt.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { useMemo } from 'react'
 import { GetTransactionReceiptResponse, ProviderInterface } from 'starknet'
 import { useStarknet } from '../providers'
 import { useInvalidateOnBlock } from './invalidate'
+import { Chain, useNetwork } from '..'
```
Comment (fracek) [inline packages/core/src/hooks/receipt.ts:6]: Maybe it should be `./network`.

File: packages/core/src/hooks/transaction.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import { useQueries, useQuery } from '@tanstack/react-query'
 import { GetTransactionResponse, ProviderInterface } from 'starknet'
 import { useStarknet } from '../providers'
 import { useInvalidateOnBlock } from './invalidate'
+import { useNetwork } from './network'
+import { Chain } from '..'
```
Comment (fracek) [inline packages/core/src/hooks/transaction.ts:6]: Maybe also from `./network`.

File: packages/core/src/providers/starknet.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ export interface StarknetState {
 // as defaultProvider from starknet.js is initialized to testnet2
 const customDefaultProvider = new Provider({
   sequencer: {
-    network: 'goerli-alpha',
+    network: constants.NetworkName.SN_GOERLI,
```
Comment (fracek) [inline packages/core/src/providers/starknet.tsx:42]: Should we default to mainnet for v1?

**Review by fracek — COMMENTED** [review]
I quickly went through it and it looks good. Will review it one more time then I'll merge it as soon as the CI passes.

File: packages/core/src/providers/starknet.tsx (hunk 3 above)
Comment (dhruvkelawala) [reply packages/core/src/providers/starknet.tsx:42]: Good idea

**Comment by fracek** [comment]
Thanks! Merged into a new feature branch for v1.

**Pull request merged**